javascript/swiggy.js

```diff
@@ -0,0 +1,158 @@
+            "lineItems": [{
+                "id": "1121",
+                "name": "Plain Veg Biryani",
+                "quantity": 2,
+                "price": 149,
+                "currency": "INR"
+            }],
+            "shippingFee": 0,
+            "discount": 0,
+            "tax": 0,
+            "subTotal": 149
+        }
+    }
+
+    const cartItems = fetchCartItems();
+
+    const itemsByCategoryMap = new Map();
+
+	function addItemByCategoryInMap(key,element){
+		if (!itemsByCategoryMap.has(key)) {
+			itemsByCategoryMap.set(key, []);
+		}
+		itemsByCategoryMap.get(key).push(element);
+	}
+	function makeItemsByCategoryMap(element){
+			element.categories.forEach(key => addItemByCategoryInMap(key,element));
+	}
+	
+    menuItems.forEach(element => makeItemsByCategoryMap(element));
+
+    const main = document.querySelector("main");
+
+    const categoriesContainer = document.querySelector(".categories");
+
+    const categoriesList = document.createElement("ul");
+	function addListItemInCategoryList(element){
+        const categoryName = document.createElement("li");
+        categoryName.id = element.id;
+        categoryName.innerText = element.displayName;
+        categoriesList.append(categoryName);
+    }
+    categories.forEach(element => addListItemInCategoryList(element));
+    categoriesContainer.append(categoriesList);
+
+    const menuItemsContainer = document.querySelector(".menu-items");
+
+    function addCategoryInfo(key) {
+        return `<h3>${key}</h3>
+		<p>${itemsByCategoryMap.get(key).length} Items</p>`
+    }
+
+    function addItemDescription(key, index) {
+        return `<div>
+		<img src="images/veg_mark.png" alt="Veg Mark Logo" class="veg-mark">
+		<p class="dish-name">${itemsByCategoryMap.get(key)[index].displayName}</p>
+		<p> &#8377 ${itemsByCategoryMap.get(key)[index].price}</p>
+		</div>
+		<div class="container">
+		<img src=${itemsByCategoryMap.get(key)[index].imgUrl} alt="food Image" class=item-image>
+		<button class="add-item-btn">ADD</button>
+		</div>`;
+    }
+
+    function addItemToCartegory(key, index) {
+        const itemContainer = document.createElement("div");
+        itemContainer.className = "item-description";
+        itemContainer.innerHTML = addItemDescription(key, index);
+
+        menuItemsContainer.append(itemContainer);
+    }
+
+    function addCategoryToMenuItems(key) {
+        const categoryInfoContainer = document.createElement("div");
+        categoryInfoContainer.className = "category-info";
+        categoryInfoContainer.innerHTML = addCategoryInfo(key);
+        menuItemsContainer.append(categoryInfoContainer);
+        for (let i = 0; i < itemsByCategoryMap.get(key).length; i++) {
+            addItemToCartegory(key, i);
+        }
+    }
+
+    for (const [key, value] of itemsByCategoryMap.entries()) {
+        addCategoryToMenuItems(key);
+    }
+
+    const cartContainer = document.querySelector(".cart");
+
+    function addCartDescription() {
```

Comment:
`addCartDescription` <> `createCartDescriptionTemplate`

Move all the templates together under // Templates so that it's easier to locate templates at once place. And rewrite this template function as per the comment above